### PR TITLE
Terraform modules to create EKS cluster with autoscaling capability

### PR DESCRIPTION
**Pull request by JFer11**
### Description

This pull request adds Terraform configuration to create an Amazon Elastic Kubernetes Service (EKS) cluster with a managed node group. The configuration consists of two parts. The first part creates a VPC that includes public and private subnets, NAT gateways, and DNS resolution. The second part creates an EKS cluster that is provisioned in the previously created VPC and deploys two managed node groups, one for general use and the other for running Spot Instances. Additionally, the configuration creates an IAM role for the cluster autoscaler with the necessary policies to work in the EKS cluster, a Kubernetes service account for the cluster autoscaler, and a kubectl provider to interact with the EKS cluster.

### Modules and Resources

- The provider block specifies the AWS provider configuration to create resources on AWS, such as the region and profile to use.

- The terraform block sets the Terraform version and defines the required providers that the configuration needs to work. The configuration uses two providers, the kubectl provider, and the helm provider.

- The module "vpc" block defines the module that creates the VPC resources, including subnets, NAT gateways, and DNS resolution.

- The module "eks" block defines the module that creates the EKS cluster and the managed node groups. It also defines the node groups' instance types, capacity types, and scaling policies.

- The data "aws_eks_cluster" "default" and data "aws_eks_cluster_auth" "default" blocks retrieve the EKS cluster's endpoint and CA certificate data and authenticate kubectl with the EKS cluster.

- The module "cluster_autoscaler_irsa_role" block defines the IAM role for the cluster autoscaler and attaches the necessary policies to work with the EKS cluster.

- The provider "kubectl" block configures the kubectl provider to connect to the EKS cluster using the aws command-line tool and the fer-libera profile.

- Finally, the resource "kubectl_manifest" "service_account" block creates a Kubernetes service account for the cluster autoscaler, which is used to run the cluster-autoscaler pod on the EKS cluster.

### CODE REVIEW

File: terraform/2-eks.tf [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      protocol                      = "tcp"
+      from_port                     = 9443
+      to_port                       = 9443
+      source_cluster_security_group = true
+      description                   = "Allow access from control plane to webhook port of AWS load balancer controller"
+    }
+  }
+
+  tags = {
+    Environment = "staging"
```
Comment (jstaricco10) [inline terraform/2-eks.tf]: Should be "production" .......

File: delete-local-cluster.sh
Comment (jopealgorta) [inline delete-local-cluster.sh:1]: Maybe move these files to a `scripts` folder.

**Review by jopealgorta — APPROVED** [review]
Don't understand much about the specidfics of Terraform, but tested locally and looks good.

File: delete-local-cluster.sh
Comment (JFer11) [reply delete-local-cluster.sh:1]: Good idea, I'll include it in the last PR.

**Pull request merged**